pkg/cmd/stack/wait.go

```diff
@@ -0,0 +1,191 @@
+	"github.com/okteto/okteto/pkg/k8s/deployments"
+	"github.com/okteto/okteto/pkg/k8s/jobs"
+	"github.com/okteto/okteto/pkg/k8s/statefulsets"
+	"github.com/okteto/okteto/pkg/model"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func Wait(ctx context.Context, stackName, svcName, namespace string) error {
+
+	if stackName == "" {
+		return fmt.Errorf("Invalid command: stack name must is required.")
+	}
+	if svcName == "" {
+		return fmt.Errorf("Invalid command: service name is required.")
+	}
+
+	c, _, err := client.GetLocal()
+	if err != nil {
+		return err
+	}
+
+	configMap, err := configmaps.Get(ctx, model.GetStackConfigMapName(stackName), namespace, c)
+	if err != nil {
+		return fmt.Errorf("Invalid command: stack '%s' does not exist on namespace '%s'.", stackName, namespace)
+	}
+
+	s, err := translateConfigMapToStack(configMap)
+	if err != nil {
+		return err
+	}
+
+	if svc, ok := s.Services[svcName]; ok {
+		addHiddenExposedPortsToDependentSvcs(ctx, s, svc)
+		err = waitForSvcs(ctx, s, svc, c)
+		if err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("Invalid command: stack '%s' does not exist.", stackName)
+	}
+
+	return nil
+}
+
+func translateConfigMapToStack(configMap *apiv1.ConfigMap) (*model.Stack, error) {
+	isCompose, err := strconv.ParseBool(configMap.Data[ComposeField])
+	if err != nil {
+		return nil, err
+	}
+
+	decoded, err := base64.StdEncoding.DecodeString(configMap.Data[YamlField])
+	if err != nil {
+		return nil, err
+	}
+
+	stack, err := model.ReadStack(decoded, isCompose)
+	if err != nil {
+		return nil, err
+	}
+	stack.Name = configMap.Data[NameField]
+	stack.Namespace = configMap.Namespace
+	return stack, nil
+}
+
+func addHiddenExposedPortsToDependentSvcs(ctx context.Context, s *model.Stack, svc *model.Service) {
+	for dependentSvcName := range svc.DependsOn {
+		addHiddenExposedPortsToSvc(ctx, s.Services[dependentSvcName], s.Namespace)
+	}
+}
+
+func waitForSvcs(ctx context.Context, stack *model.Stack, svc *model.Service, c kubernetes.Interface) error {
+	var err error
+	for dependentSvc, condition := range svc.DependsOn {
+		fmt.Printf("Waiting for service '%s'\n", dependentSvc)
+		switch condition.Condition {
+		case model.DependsOnServiceRunning:
+			err = waitForSvcRunning(ctx, stack, dependentSvc, c)
+			if err != nil {
+				return err
+			}
+		case model.DependsOnServiceHealthy:
+			err = waitForSvcHealthy(stack, dependentSvc)
+			if err != nil {
+				return err
+			}
+		case model.DependsOnServiceCompleted:
+			err = waitForSvcCompleted(ctx, stack, dependentSvc, c)
+			if err != nil {
+				return err
+			}
+		}
+		fmt.Printf("Service '%s' is ready\n", dependentSvc)
+	}
+	return nil
+}
+
+func waitForSvcRunning(ctx context.Context, stack *model.Stack, svcName string, c kubernetes.Interface) error {
+	svcToWaitFor := stack.Services[svcName]
+
+	ticker := time.NewTicker(100 * time.Millisecond)
+	timeout := time.Now().Add(300 * time.Second)
```

Comment:
One option is to print a log a fail. Kubernetes will retry the init container with the right backlogs